### ISSUE TITLE
webpack-dev-server not updating file which is "out of context"

### ISSUE THREAD

**Issue by peter-perot** [webpack/webpack#2580]
I have read myriads of tutorials, but nothing did help me to get webpack-dev-server working properly on a Windows 10 development machine.

#### *TL;DR:* It seems that webpack-dev-server only watches the filesystem under the *context node*, so if *context* is `"./src"`, the folder `"./dist"` (which is not a subfolder of `"./src"`) is not watched. ####

I have a very simple React application. The directory structure looks like this:

````
package.json
webpack.config.js

+ src\
  main.jsx
  favicon.ico
  indexTempl.html

+ src\
      + components\
        homePage.jsx  

+ src\
      + components\
                   + about\
                     aboutPage.jsx

+ src\
      + components\
                   + common\
                     header.jsx

+ src\
      + images\
        logo.png
````

As you can see: a pretty simple React application.

My `package.json`:

````javascript
{
  "name": "react-flux",
  "version": "0.0.1",
  "description": "react-flux",
  "main": "webpack.config.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "webpack-dev-server --progress --inline --hot"
  },
  "author": "perot",
  "license": "ISC",
  "private": true,
  "devDependencies": {
    "babel-core": "^6.9.1",
    "babel-loader": "^6.2.4",
    "babel-preset-es2015": "^6.9.0",
    "babel-preset-react": "^6.5.0",
    "babel-preset-react-hmre": "^1.1.1",
    "copy-webpack-plugin": "^3.0.1",
    "file-loader": "^0.8.5",
    "html-webpack-plugin": "^2.19.0",
    "webpack": "^1.13.1",
    "webpack-dev-server": "^1.14.1"
  },
  "dependencies": {
    "jquery": "^2.2.4",
    "react": "^15.1.0",
    "react-dom": "^15.1.0"
  }
}
````

My `webpack.config.js`:

````javascript
var CopyWebpackPlugin = require("copy-webpack-plugin");
var HtmlWebpackPlugin = require("html-webpack-plugin");

module.exports = {
  context: __dirname + "/src",

  entry: {
    javascript: "./main.jsx"
  },

  plugins: [
    new CopyWebpackPlugin([{ from: "images/*" }]),
    new HtmlWebpackPlugin({
      hash: true,
      filename: "./index.html",
      template: "./indexTempl.html"
    })
  ],

  output: {
    filename: "app.js",
    path: "./dist"
  },

  resolve: {
    extensions: ["", ".js", ".jsx"]
  },

  module: {
    loaders: [
      {
        test: /\.jsx?$/,
        exclude: /node_modules/,
        loaders: ["babel?presets[]=es2015,presets[]=react,presets[]=react-hmre"]
      }
    ]
  }
};
````

Now the problem:

When I'm editing the `indexTempl.html`, webpack realizes the change and begins with rebundling. Now when I refresh my webpage in the browser, the changes can be seen. That's okay so far.

But when I'm editing the `homePage.jsx`, webpack does not get notified about the change - nothing changes when I refresh the page in the browser.

**Now I made an interesting observation:**

When I delete a single line in `webpack.config.js`, it's working - but not the way I like.

Deleting the line

`context: __dirname + "/src",`

and now putting a `"src/"` in every path of my configuration solves the problem.

But it's pretty ugly: Now I have to write `"http://localhost:8080/src/"` instead of `"http://localhost:8080"`, and some links are boken.

**However: It seems that webpack-dev-server only watches the filesystem under the *context node*, so if *context* is `"./src"`, the folder `"./dist"` (which is not a subfolder of `"./src"`) is not watched.**

Does anybody know what is wrong in my configuration?

**Comment by peter-perot** [comment]
**Found a solution.** The problem is: Combining pathes like `__dirname + "/directory"` does not work (anybody knows why?). Instead I have to use `Path.resolve(__dirname, "directory")`.

Here comes my working configuration:

````javascript
var CopyWebpackPlugin = require("copy-webpack-plugin");
var HtmlWebpackPlugin = require("html-webpack-plugin");
var Path = require("path");

module.exports = {
    // !!!! use Path.resolve(...) !!!!
    context: Path.resolve(__dirname, "src"),

    entry: {
        javascript: "./main.jsx"
    },

    plugins: [
        new CopyWebpackPlugin([{ from: "images/*" }]),
        new HtmlWebpackPlugin({
            hash: true,
            filename: "./index.html",
            template: "./indexTempl.html"
        })
    ],

    output: {
        filename: "app.js",

        // !!!! use Path.resolve(...) !!!!
        context: Path.resolve(__dirname, "dist"),

        publicPath: "/"
    },

    resolve: {
        extensions: ["", ".js", ".jsx"]
    },

    module: {
        loaders: [
            {
                test: /\.jsx?$/,
                exclude: /node_modules/,
                loaders: ["babel?presets[]=es2015,presets[]=react,presets[]=react-hmre"]
            }
        ]
    }
};
````

I just replaced `context: __dirname + "/src"` by `context: Path.resolve(__dirname, "src")` and `context: __dirname + "/dist"` by `context: Path.resolve(__dirname, "dist")`, and this way it works.

**Anybody knows what the difference between using "+" (string concatenation operator) and `Path.resolve()` is?**
